src/caffe/layers/cudnn_batch_norm_layer.cu

```diff
@@ -0,0 +1,112 @@
+#ifdef USE_CUDNN
+#include <algorithm>
+#include <cfloat>
+#include <vector>
+
+#include "thrust/device_vector.h"
+
+#include "caffe/layers/cudnn_batch_norm_layer.hpp"
+#include "caffe/util/math_functions.hpp"
+
+namespace caffe {
+
+template <typename Dtype>
+void CuDNNBatchNormLayer<Dtype>::Forward_gpu(
+    const vector<Blob<Dtype>*>& bottom,
+    const vector<Blob<Dtype>*>& top) {
+  const Dtype* bottom_data = bottom[0]->gpu_data();
+  const Dtype* scale_data = this->blobs_[0]->gpu_data();
+  const Dtype* bias_data = this->blobs_[1]->gpu_data();
+
+  Dtype* top_data = top[0]->mutable_gpu_data();
+  Dtype* save_mean = save_mean_.mutable_gpu_data();
+  Dtype* save_inv_var = save_inv_var_.mutable_gpu_data();
+  double epsilon = max(this->eps_, CUDNN_BN_MIN_EPSILON);
+
+  if (this->phase_ == TRAIN) {
+    // Call Batch normalization forward
+    CUDNN_CHECK(cudnnBatchNormalizationForwardTraining(
+      Caffe::cudnn_handle(),
+      mode_,
+      cudnn::dataType<Dtype>::one,
+      cudnn::dataType<Dtype>::zero,
+      bottom_desc_,
+      bottom_data,
+      bottom_desc_,
+      top_data,
+      scale_bias_mean_var_desc_,
+      scale_data,
+      bias_data,
```

Comment:
Please don't add all these linebreaks unless you think there's a really good reason for it; it doesn't agree with existing style, and it makes the code harder to read by stretching it out. It's okay to group arguments logically.